### PR TITLE
futures: fix tests not compiling on some feature combinations

### PR DESCRIPTION
**Pull request by hawkw**
## Motivation

Previously, the tests for `std::future` support were always compiled,
regardless of whether or not the `std::future` feature was enabled. This
is because the tests were in the integration tests dir, and were not
feature flagged.

Additionally, when disabling the `std` feature, `tracing-futures` tests 
will fail to compile, as the test support library depends on `std`.

## Solution

I've moved these tests into modules in the lib tests, so that they can
only be compiled when the appropriate feature flags are enabled. I've
also made some tweaks to the test support code to accommodate this.

I've also made the `tests` module be conditional on the presence of the
`std` feature. Ideally, we would be able to run tests for the subset of
functionality available on `no-std` when the feature is disabled, but
there are some obstacles to doing this easily.

This fixes some of the issues described in #632.

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

File: tracing-futures/src/tests/mod.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#[cfg_attr(
+    not(feature = "std-future"),
+    path = "../../../tracing/tests/support/mod.rs"
+)]
+// We don't need all the test support code that other crates need, so don't
+// warn us if it's unused.
+#[allow(dead_code, unreachable_pub)]
+mod support;
```
Comment (hawkw) [inline tracing-futures/src/tests/mod.rs]: apparently something in here is breaking rustfmt, may have to figure out a different solution...

**Comment by jplatte** [comment]
Bump. Just ran into tracing-futures not compiling with `--no-default-features` (when trying to verify I didn't break things with #1185).

**Comment by bryangarza** [comment]
Let's rebase and merge this one?

**Comment by hawkw** [comment]
Closing this as I believe it's obsolete and the issue may have already been fixed separately.

**Pull request closed without merging**